scripts/reboot_smartswitch_helper

```diff
@@ -0,0 +1,293 @@
+    fi
+}
+
+# Function to rescan PCI module
+function pci_reattach_module()
+{
+    local DPU_NAME=$1
+    local DPU_BUS_INFO=$2
+    python3 -c "from reboot_helper import pci_reattach_module; pci_reattach_module('${DPU_NAME}')"
+    if [ $? -ne 0 ]; then
+        log_message "Error: PCI reattach vendor API is not available"
+        echo 1 > /sys/bus/pci/rescan
+    fi
+}
+
+# Function to reboot DPU
+function reboot_dpu_platform()
+{
+    local DPU_NAME=$1
+    local REBOOT_TYPE=$2
+    python3 -c "import reboot_helper; reboot_helper.reboot_dpu('${DPU_NAME}', '${REBOOT_TYPE}')"
+}
+
+# Function to wait for DPU reboot status
+function wait_for_dpu_reboot_status()
+{
+    local dpu_ip=$1
+    local port=$2
+
+    if [[ -z "$PLATFORM_JSON_PATH" ]]; then
+        log_message "Error: PLATFORM_JSON_PATH is not defined"
+        exit $EXIT_ERROR
+    fi
+
+    local dpu_halt_services_timeou
+    dpu_halt_services_timeout=$(jq -r '.dpu_halt_services_timeout' "$PLATFORM_JSON_PATH" 2>/dev/null)
+    if [ $? -ne 0 ]; then
+        log_message "Error: Failed to retrieve dpu_halt_services_timeout from ${PLATFORM_JSON_PATH}"
+        return ${EXIT_ERROR}
+    fi
+
+    local poll_interval=5
+    local waited_time=0
+    while true; do
+        local reboot_status
+        get_reboot_status "${dpu_ip}" "${port}"
+        reboot_status=$?
+        if [ $reboot_status -eq ${EXIT_SUCCESS} ]; then
+            break
+        fi
+
+        sleep "$poll_interval"
+        waited_time=$((waited_time + poll_interval))
+        if [ $waited_time -ge $dpu_halt_services_timeout ]; then
+            log_message "Error: Timeout waiting for DPU ${DPU_NAME} to finish rebooting"
+            return ${EXIT_ERROR}
+        fi
+    done
+}
+
+# Function to send reboot command to DPU
+function gnmi_reboot_dpu()
+{
+    # Retrieve DPU IP and GNMI port
+    dpu_ip=$(get_dpu_ip "${DPU_NAME}")
+    log_message "DPU IP ${DPU_NAME}: $dpu_ip"
+    port=$(get_gnmi_port "${DPU_NAME}")
+    if [ -z "$port" ]; then
+        port=$GNMI_PORT # Default GNMI port
+    fi
+    log_message "GNMI port ${DPU_NAME}: $port"
+
+    if [ -z "$dpu_ip" ]; then
+        log_message "Error: Failed to retrieve DPU IP for ${DPU_NAME}"
+        return ${EXIT_ERROR}
+    fi
+
+    docker exec -i gnmi gnoi_client -target ${dpu_ip}:${port} -logtostderr -insecure -rpc Reboot -jsonin '{"method":3}'
+    if [ $? -ne 0 ]; then
+        log_message "Error: Failed to send reboot command to DPU ${DPU_NAME}"
+        return ${EXIT_ERROR}
+    fi
+
+    wait_for_dpu_reboot_status "${dpu_ip}" "${port}"
+}
+
+function reboot_dpu()
+{
+    local DPU_NAME=$1
+    local REBOOT_TYPE=$2
+    local DPU_INDEX=${DPU_NAME//[!0-9]/}
+
+    debug "User requested rebooting device ${DPU_NAME} ..."
+
+    # Send reboot command to DPU
+    gnmi_reboot_dpu "${DPU_NAME}"
+    if [ $? -ne 0 ]; then
+        log_message "Error: Failed to send gnoi command to reboot DPU ${DPU_NAME}"
+        return ${EXIT_ERROR}
```

Comment:
We should not interrupt the reboot if the pre-shutdown fails. 
If the pre-shutdown fails we should log error message and continue to the platform reboot

Reply:
Got it. Thanks!